tests/protocol/test_hostplugin.py

```diff
@@ -129,65 +129,169 @@ def _validate_hostplugin_args(self, args, goal_state, exp_method, exp_url, exp_d
+
+    @patch("azurelinuxagent.common.protocol.hostplugin.HostPluginProtocol.ensure_initialized",
+           return_value=True)
+    @patch("azurelinuxagent.common.protocol.wire.StatusBlob.upload",
+           return_value=True)
+    @patch("azurelinuxagent.common.protocol.hostplugin.HostPluginProtocol._put_page_blob_status",
+           side_effect=HttpError("503"))
+    @patch("azurelinuxagent.common.protocol.wire.WireClient.update_goal_state")
+    def test_fallback_channel_503(self, patch_update, patch_put, patch_upload, _):
         """
-        Validate that if host plugin fails and the direct connection fails the
-        default channel remains unchanged.
+        When host plugin returns a 503, we should fall back to the direct channel
         """
         test_goal_state = wire.GoalState(WireProtocolData(DATA_FILE).goal_state)
-        status = restapi.VMStatus(status="Ready",
-                                  message="Guest Agent is running")
-        wire.HostPluginProtocol.set_default_channel(False)
-        with patch.object(wire.HostPluginProtocol,
-                          "ensure_initialized",
-                          return_value=True):
-            with patch.object(wire.StatusBlob,
-                              "upload",
-                              return_value=False) as patch_upload:
-                with patch.object(wire.HostPluginProtocol,
-                                  "_put_page_blob_status",
-                                  side_effect=wire.HttpError("put failure")) as patch_put:
-                    self.assertFalse(wire.HostPluginProtocol.is_default_channel())
-                    client = wire.WireProtocol(wireserver_url).client
-                    client.get_goal_state = Mock(return_value=test_goal_state)
-                    client.ext_conf = wire.ExtensionsConfig(None)
-                    client.ext_conf.status_upload_blob = sas_url
-                    client.ext_conf.status_upload_blob_type = page_blob_type
-                    client.status_blob.set_vm_status(status)
-                    client.upload_status_blob()
-                    self.assertEqual(1, patch_put.call_count)
-                    self.assertEqual(1, patch_upload.call_count)
-                    self.assertFalse(wire.HostPluginProtocol.is_default_channel())
+        status = restapi.VMStatus(status="Ready", message="Guest Agent is running")
+
+        wire_protocol_client = wire.WireProtocol(wireserver_url).client
+        wire_protocol_client.get_goal_state = Mock(return_value=test_goal_state)
+        wire_protocol_client.ext_conf = wire.ExtensionsConfig(None)
+        wire_protocol_client.ext_conf.status_upload_blob = sas_url
+        wire_protocol_client.ext_conf.status_upload_blob_type = page_blob_type
+        wire_protocol_client.status_blob.set_vm_status(status)
+
+        # act
+        wire_protocol_client.upload_status_blob()
+
+        # assert direct route is not called
```

Comment:
The comment and the assert message do not appear to agree.

Reply:
Thanks, fixed.